upload/admin/controller/extension/shipping/zasilkovna.php

```diff
@@ -951,13 +949,61 @@ public function orders() {
 
 	/**
 	 * Handler for detail of Packeta order.
+	 * @throws ReflectionException
 	 */
 	public function order_detail()
 	{
-		$data = $this->initPageData(self::ACTION_ORDER_DETAIL, 'text_order_detail', ['order_id' => $this->request->get['order_id']]);
+		/** @var OrderDetailPage $orderDetailPage */
+		$orderDetailPage = $this->diContainer->get(OrderDetailPage::class);
+		$orderId = (int) $this->request->get['order_id'];
+		$this->load->language(self::ROUTING_BASE_PATH);
+
+		if ($this->request->server['REQUEST_METHOD'] === 'POST') {
+			if ($orderDetailPage->save($this->request->post)) {
+				$this->session->data['success'] = $this->language->get('order_detail_changes_saved');
+			} else {
+				$this->session->data['error_warning'] = $this->language->get('order_detail_changes_not_saved');
+			}
+
+			$this->response->redirect(
+				$this->createAdminLink(
+					self::ACTION_ORDER_DETAIL,
+					['order_id' => (int) $this->request->post['order_id']]
+				)
+			);
+		}
+
+		$this->document->addScript('https://widget.packeta.com/v6/www/js/library.js');
+		$this->document->addScript('/catalog/view/javascript/zasilkovna/shippingExtensionBack.js?v=' . Tools::MODULE_VERSION);
+		$data = $this->initPageData(self::ACTION_ORDER_DETAIL, 'text_order_detail', ['order_id' => $orderId]);
+
+		$order = $orderDetailPage->getOrderData($orderId);
+		$routeToListLink = $this->createAdminLink(self::ROUTE_PACKETA_ORDERS);
+
+		if (!$order) {
+			$this->session->data['error_warning'] = sprintf(
+				$this->language->get('order_detail_order_doesnt_exist'),
+				$orderId
+			);
+			$this->response->redirect($routeToListLink);
+		}
 
-		$this->response->setOutput($this->load->view('extension/shipping/zasilkovna_order_detail', $data));
+		$data['order'] = $order;
+		$data['link_to_list'] = $routeToListLink;
+
+		$settings = $this->getSettings();
+
+		$data['widget'] = [
+			'api_key' => $settings['shipping_zasilkovna_api_key'],
```

Comment:
Co když nebude klíč shipping_zasilkovna_api_key v settings existovat ? Nešlo by použít this->config->get('shipping_zasilkovna_api_key'), když potřebuješ jen jednu hodnotu z celého configu ?